shared/runtime/pyexec.c

```diff
@@ -140,7 +143,12 @@ STATIC int parse_compile_execute(const void *source, mp_parse_input_kind_t input
             mp_hal_stdout_tx_strn("\x04", 1);
         }
 
-        // check for SystemExit
+        #if MICROPY_SCHED_VM_ABORT
+        if (nlr.ret_val == NULL) {
```

Comment:
There is an existing comment on the definition of `ret_val` that says:

> `// always a concrete object (an exception instance)`

Perhaps this should be updated to explain the change in semantics when `MICROPY_SCHED_VM_ABORT` is enabled?

Reply:
Thanks, I've updated that comment.